sdk/textanalytics/Azure.AI.TextAnalytics/src/Validation.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.CompilerServices;
+using static Azure.AI.TextAnalytics.TextAnalyticsClientOptions;
+
+namespace Azure.AI.TextAnalytics
+{
+    internal static class Validation
```

Comment:
Can we move this to shared source so the versioning exceptions look the same across libraries?  Or even move this into the `Argument` class?

Reply:
After talking with Anne, it sounded like we didn't even want to consider client-side validation yet. I didn't want to move it to shared source prematurely, but figured we could - just like we did with the challenge auth policy - when needed.

Reply:
Per offline discussion, we'll move this when needed, but I'll make some of these other changes and add some comments with some of our future thoughts we don't need to implement right now.

Reply:
@tg-msft, why shared source and not a public type?